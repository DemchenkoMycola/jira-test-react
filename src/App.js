import React from 'react';
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import Home from './views/Home';
import Projects from './views/Projects';
import Issues from './views/Issues';
import NoMatch from './views/NoMatch';
import ProjectDetails from './views/ProjectDetails';
// import Header from './components/Header';

const App = () => {
    return (
      <main>
        <Router>
          {/* <Header /> */}
          <Switch>
            <Route exact path="/" component={Home} />
            <Route path="/projects/:userId" component={Projects} />
            <Route path="/issues/:id" component={Issues} /> 
            <Route path="/details/:id" component={ProjectDetails} />
            <Route component={NoMatch} />
          </Switch>
        </Router>
      </main>
    );
}

export default App;
