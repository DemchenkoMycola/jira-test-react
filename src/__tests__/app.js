import React from 'react';
import App from '../App'
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

describe('app', () =>{
    const store = createStore(() => ({}));

    it('render without crashing', () =>{
       const app = renderer.create(
            <Provider store={store}>
                <App />
            </Provider>)
            .toJSON();
        expect(app).toMatchSnapshot()
    });

});