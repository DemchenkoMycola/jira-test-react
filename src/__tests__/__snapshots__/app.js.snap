// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app render without crashing 1`] = `
<main>
  <section
    className="auth_section"
  >
    <nav>
      <ul
        className="auth_nav"
      >
        <li
          className="auth_item active"
          onClick={[Function]}
        >
          Login
        </li>
        <li
          className="auth_item "
          onClick={[Function]}
        >
          Register
        </li>
      </ul>
    </nav>
    <section
      className="register_form"
    >
      <legend>
        Register
      </legend>
      <span
        className="errors"
      >
        
      </span>
      <span
        className="messages"
      >
        
      </span>
      <label>
        Username
      </label>
      <input
        className="input"
        name="username"
        onChange={[Function]}
        placeholder="username"
        type="text"
        value=""
      />
      <label>
        Email
      </label>
      <input
        className="input"
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
      <label>
        Password
      </label>
      <input
        className="input"
        name="passowrd"
        onChange={[Function]}
        placeholder="password"
        type="password"
        value=""
      />
      <button
        onClick={[Function]}
      >
        register
      </button>
    </section>
  </section>
</main>
`;
